### ISSUE TITLE
How to implement something like `tf.nn.moments()`?

### ISSUE THREAD

**Issue by reiinakano** [PAIR-code/deeplearnjs#138]
Basically, I want to be able to calculate the mean + variance across different axes just like `tf.nn.moments` in Tensorflow. This is so I could implement instance normalization (normalization across height and width in images).

I'm a bit stuck on the best way to calculate means and variances over various axes in an Array3D (I'd like one mean and one variance per channel). Could you point me to some existing methods and functions I could use to implement this? Thanks.

**Comment by nsthorat** [comment]
Just wanted to let you know we're making progress on this. @dsmilkov has been hard at work adding batching, broadcasting, and axis-aligned ops at the compiler level:
https://github.com/PAIR-code/deeplearnjs/pull/219

He's working on some more functionality needed for moments on the GPU, we'll update this thread when it's finally ready, but wanted to let you know that we haven't forgotten about this :)

**Comment by reiinakano** [comment]
@nsthorat no worries, I've been checking the deeplearnjs commits from time to time as well. Looks like you're pretty close. Thanks for the update!

**Comment by dsmilkov** [comment]
Fixed in #277. @reiinakano let us know when you update the style transfer demo and if you run into any problems. Cheers!

**Comment by reiinakano** [comment]
Nice, thanks! I'll wait for the next release then update it when I have time.

**Comment by dsmilkov** [comment]
Great! Released `0.3.4` with `math.moments`
